tests/common.go

```diff
@@ -5131,3 +5131,68 @@ func GetPoolIDsFromVolName(volName string) ([]string, error) {
 	}
 	return poolUuids, err
 }
+
+// WaitTillEnterMaintenance Wait till Node Enters maintenance mode
+func WaitTillEnterMaintenanceMode(n node.Node) {
+	t := func() (interface{}, bool, error) {
+		if Inst().V.IsNodeInMaintenance(n) == true {
+			return nil, true, nil
+		}
+		return nil, false, fmt.Errorf("Not in Maintenance mode")
+	}
+
+	_, err := task.DoRetryWithTimeout(t, 20*time.Minute, 2*time.Minute)
+	log.FailOnError(err, fmt.Sprintf("Failed to Enter maintenence mode on node [%s]", n.Name))
+}
+
+// ExitMaintenanceMode waits till Node exits from Maintenance Mode
+func ExitFromMaintenanceMode(n node.Node) {
+	log.InfoD(fmt.Sprintf("Exiting maintenence mode on Node %s", n.Name))
+	t := func() (interface{}, bool, error) {
+		if err := Inst().V.ExitMaintenance(n); err != nil {
+			if Inst().V.IsNodeOutOfMaintenance(n) == true {
+				return nil, true, nil
+			}
+			return nil, true, err
+		}
+		return nil, false, nil
+	}
+	_, err := task.DoRetryWithTimeout(t, 15*time.Minute, 2*time.Minute)
+	log.FailOnError(err, fmt.Sprintf("Fail to exit maintenence mode on Node %s", n.Name))
```

Comment:
Same, have this function return and handle `log.FailOnError()` where you call it

Reply:
fixed